Generated enum constants collide with type names for none values

## 问题

`kb-cloud-client-go` 最新生成代码中，部分 enum 的 `none` value 会生成出与 type 完全同名的 const，导致 Go 编译失败。

## 复现方式

在 `kb-cloud-client-go` 仓库执行：

```bash
go test ./api/kbcloud
```

## 典型错误

```text
api/kbcloud/model_ai_agent_action_branch_status.go:17:2: AiAgentActionBranchStatus redeclared in this block
api/kbcloud/model_ai_agent_confidence.go:20:2: AiAgentConfidence redeclared in this block
api/kbcloud/model_ai_agent_recover_action_type.go:24:2: AiAgentRecoverActionType redeclared in this block
api/kbcloud/model_ai_agent_root_cause_type.go:19:2: AiAgentRootCauseType redeclared in this block
```

## 生成结果示例

当前生成结果类似：

```go
type AiAgentConfidence string

const AiAgentConfidence AiAgentConfidence = "none"
```

`const AiAgentConfidence` 与 `type AiAgentConfidence` 同名，因此同一个 package 内无法编译。

## 影响范围

最新 `kb-cloud-client-go/api/kbcloud` 无法编译，会阻塞依赖正式 SDK 的调用方。例如 apecloud/apecloud #18093 需要用最新 SDK 覆盖 MongoDB DMS E2E，当前只能依赖临时手改生成物验证方向，不能作为最终可合入方案。

## 初步判断

这不是 MongoDB API 语义问题。需要由 client-go 负责人判断修复位置：

- generator 对 enum varname 做去重/保留名处理；或
- OpenAPI schema 为这些 enum 提供稳定的 enum varnames / vendor extension。

--- a/.generator/src/generator/formatter.py
+++ b/.generator/src/generator/formatter.py
@@ -132,7 +132,7 @@ def get_enum_varname(schema, index):
         varname = "Empty"
     if varname == "*":
         varname = "All"
-    if varname == "None":
+    if varname.lower() == "none":
         varname = "Non"
     return camel_case(varname.replace("-", "_"))
 

--- a/.generator/src/generator/templates/api.j2
+++ b/.generator/src/generator/templates/api.j2
@@ -22,6 +22,7 @@ type {{ classname }} {{ common_package_name }}.Service
 {%- set returnType = operation|return_type %}
 {%- set formParameter = operation|form_parameter %}
 {%- set needBodyParameter = operation|need_body_parameter %}
+{%- set errorResponses = operation|responses_by_types|list %}
 {%- set operationId = operation.operationId|upperfirst %}
 
 {%- for name, parameter in operation|parameters if not parameter.required and parameter.in != "path" %}
@@ -321,14 +322,14 @@ localVarQueryParams.Add("{{ parameter.name }}", {{ common_package_name }}.Parame
 	if err != nil || localVarHTTPResponse == nil {
 		return {% if returnType %}localVarReturnValue, {% endif %}localVarHTTPResponse, err
 	}
-	{%- if returnType != '_io.Reader' %}
+	{%- if returnType != '_io.Reader' and (returnType or errorResponses) %}
 
 	localVarBody, err := common.ReadBody(localVarHTTPResponse)
 	if err != nil {
 		return {% if returnType %}localVarReturnValue, {% endif %}localVarHTTPResponse, err
 	}
 	{%- endif %}
-	{%- for responseType, (response, responseCodes) in operation|responses_by_types %}
+	{%- for responseType, (response, responseCodes) in errorResponses %}
 	{%- if loop.first %}
 
 	if localVarHTTPResponse.StatusCode >= 300 {

--- a/api/kbcloud/model_ai_agent_action_branch_status.go
+++ b/api/kbcloud/model_ai_agent_action_branch_status.go
@@ -14,7 +14,7 @@ type AiAgentActionBranchStatus string
 
 // List of AiAgentActionBranchStatus.
 const (
-	AiAgentActionBranchStatus                    AiAgentActionBranchStatus = "none"
+	AiAgentActionBranchStatusNon                 AiAgentActionBranchStatus = "none"
 	AiAgentActionBranchStatusPendingConfirmation AiAgentActionBranchStatus = "pending_confirmation"
 	AiAgentActionBranchStatusApproved            AiAgentActionBranchStatus = "approved"
 	AiAgentActionBranchStatusExecuting           AiAgentActionBranchStatus = "executing"
@@ -25,7 +25,7 @@ const (
 )
 
 var allowedAiAgentActionBranchStatusEnumValues = []AiAgentActionBranchStatus{
-	AiAgentActionBranchStatus,
+	AiAgentActionBranchStatusNon,
 	AiAgentActionBranchStatusPendingConfirmation,
 	AiAgentActionBranchStatusApproved,
 	AiAgentActionBranchStatusExecuting,

--- a/api/kbcloud/model_ai_agent_confidence.go
+++ b/api/kbcloud/model_ai_agent_confidence.go
@@ -17,14 +17,14 @@ const (
 	AiAgentConfidenceHigh   AiAgentConfidence = "high"
 	AiAgentConfidenceMedium AiAgentConfidence = "medium"
 	AiAgentConfidenceLow    AiAgentConfidence = "low"
-	AiAgentConfidence       AiAgentConfidence = "none"
+	AiAgentConfidenceNon    AiAgentConfidence = "none"
 )
 
 var allowedAiAgentConfidenceEnumValues = []AiAgentConfidence{
 	AiAgentConfidenceHigh,
 	AiAgentConfidenceMedium,
 	AiAgentConfidenceLow,
-	AiAgentConfidence,
+	AiAgentConfidenceNon,
 }
 
 // GetAllowedValues returns the list of possible values.

--- a/api/kbcloud/model_ai_agent_recover_action_type.go
+++ b/api/kbcloud/model_ai_agent_recover_action_type.go
@@ -21,7 +21,7 @@ const (
 	AiAgentRecoverActionTypeViewReport            AiAgentRecoverActionType = "view_report"
 	AiAgentRecoverActionTypeRegenerateReport      AiAgentRecoverActionType = "regenerate_report"
 	AiAgentRecoverActionTypeContinueInvestigation AiAgentRecoverActionType = "continue_investigation"
-	AiAgentRecoverActionType                      AiAgentRecoverActionType = "none"
+	AiAgentRecoverActionTypeNon                   AiAgentRecoverActionType = "none"
 )
 
 var allowedAiAgentRecoverActionTypeEnumValues = []AiAgentRecoverActionType{
@@ -32,7 +32,7 @@ var allowedAiAgentRecoverActionTypeEnumValues = []AiAgentRecoverActionType{
 	AiAgentRecoverActionTypeViewReport,
 	AiAgentRecoverActionTypeRegenerateReport,
 	AiAgentRecoverActionTypeContinueInvestigation,
-	AiAgentRecoverActionType,
+	AiAgentRecoverActionTypeNon,
 }
 
 // GetAllowedValues returns the list of possible values.

--- a/api/kbcloud/model_ai_agent_root_cause_type.go
+++ b/api/kbcloud/model_ai_agent_root_cause_type.go
@@ -16,13 +16,13 @@ type AiAgentRootCauseType string
 const (
 	AiAgentRootCauseTypeConfirmed AiAgentRootCauseType = "confirmed"
 	AiAgentRootCauseTypeCandidate AiAgentRootCauseType = "candidate"
-	AiAgentRootCauseType          AiAgentRootCauseType = "none"
+	AiAgentRootCauseTypeNon       AiAgentRootCauseType = "none"
 )
 
 var allowedAiAgentRootCauseTypeEnumValues = []AiAgentRootCauseType{
 	AiAgentRootCauseTypeConfirmed,
 	AiAgentRootCauseTypeCandidate,
-	AiAgentRootCauseType,
+	AiAgentRootCauseTypeNon,
 }
 
 // GetAllowedValues returns the list of possible values.

